azimuth/modules/base_classes/module.py

```diff
@@ -52,7 +53,8 @@ def get_effective_arguments(self) -> ModuleEffectiveArguments:
         # model_contract_method_name are excluded too because it's already in the task_name.
         return ModuleEffectiveArguments(
             mod_options=self.mod_options.dict(
-                exclude={"indices", "model_contract_method_name"}, include=self.allowed_mod_options
+                exclude={"indices", "model_contract_method_name"},
+                include=self.allowed_mod_options.union(self.required_mod_options),
```

Comment:
I see in most places you used the operator `|` for the union. I don't have a preference for one or for the other, but I guess we should keep it consistent.
```suggestion
                include=self.allowed_mod_options | self.required_mod_options,
```

Reply:
Right!! I forgot about it. Thanks!